pageserver/src/tenant/timeline.rs

```diff
@@ -2070,6 +2071,16 @@ impl Timeline {
     }
 
     fn try_spawn_size_init_task(self: &Arc<Self>, lsn: Lsn, ctx: &RequestContext) {
+        let state = self.current_state();
+        if matches!(
+            state,
+            TimelineState::Broken { .. } | TimelineState::Stopping
+        ) {
+            // Can happen when timeline detail endpoint is used when deletion is ongoing.
+            warn!("Logical size calculation was attempted to be spawned for inactive timeline");
+            return;
+        }
```

Comment:
This seems like a separate PR?
Is the `utils.py` change due to this?

certainly, the `neon_fixtures.py` change is due to this.

Do we really need to log something here? We have the same time-of-check-vs-time-of-use situation with `collect_metrics_iteration` .

Can we make `get_current_logical_size` fail with a distinguished error instead, that says "shutting down"?
How would the callers handle that:

* `collect_metrics_iteration`
  * skip without making noise
* `build_timeline_info_common`
  * it currently logs an error but continues to produce a response. We should fix that anyways.
* `handle_walreceiver_connection`
  * hang up, we're shutting down

Reply:
It triggered noticeable amount of flakiness with this PR. I think if timeline is broken or is being stopped we're ok with receiving approximate size. Stopped timelines should be also excluded from any consumption related stuff.

Note that this can only happen if this is called shortly after ps restart.

> Can we make get_current_logical_size fail with a distinguished error instead, that says "shutting down"?

TimelineState::Broken doesnt mean that we're shutting down.

> Do we really need to log something here?

We can consider it is a norm and just remove the message.

***

I can move it to separate PR if you want

Reply:
> We can consider it is a norm and just remove the message.

I'm fine with that.

> I can move it to separate PR if you want

I think it would make this PR somewhat clearer. Since this PR uncovered the problem though, maybe it's better to keep them in one PR. Just mention in the PR description / commit message that this was a pre-existing issue that your main change uncovered.

Reply:
Removed the message (resolving to clearly see remaining items, feel free to reopen)